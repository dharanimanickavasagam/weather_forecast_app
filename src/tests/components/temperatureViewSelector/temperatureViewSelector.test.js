import React from "react";
import { shallow } from "enzyme";
import TemperatureViewSelector from "../../../components/temperatureViewSelector/temperatureViewSelector";
import CurrentTemperature from "../../../components/currentTemperature/currentTemperature";
import FiveDayTemperature from "../../../components/fiveDayTemperature/fiveDayTemperature";
let wrapper, tabData;

beforeAll(() => {
  tabData = [
    {
      component: <CurrentTemperature />,
      label: "Current Temperature"
    },
    {
      component: <FiveDayTemperature />,
      label: "5 Day Forecast"
    }
  ];
  wrapper = shallow(<TemperatureViewSelector tabData={tabData} />);
});

describe("match snapshots ", () => {
  it("match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("test if the components are rendered ", () => {
  it("should render the TabSelector component", () => {
    expect(wrapper.find("TabSelector")).toBeTruthy();
  });
});
