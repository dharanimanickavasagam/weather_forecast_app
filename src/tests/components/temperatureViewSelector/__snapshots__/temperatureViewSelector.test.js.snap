// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshots  match snapshot 1`] = `
<div
  className="tab"
>
  <TabSelector
    tabData={
      Array [
        Object {
          "component": <CurrentTemperature />,
          "label": "Current Temperature",
        },
        Object {
          "component": <FiveDayTemperature />,
          "label": "5 Day Forecast",
        },
      ]
    }
  />
</div>
`;
